Add action build

iBoonie opened the pull request:
May as well add action building for this.

iBoonie committed:
Create ci.yml

chrb22 commented:
Thanks. I wanted to add this, but didn't know my way around GitHub Actions. Does this also automatically upload the builds together with releases?

chrb22 commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,194 @@
+          name: sendvaredit_${{ runner.os }}
+          path: sendvaredit/build/package
+      
+      - name: Upload artifact
+        if: github.event_name == 'push' && strategy.job-index == 0
+        uses: actions/upload-artifact@v4
+        with:
+          name: sendvaredit_versioning_files
+          path: sendvaredit/build/includes
```
Also what is this for? I have never had a build/includes folder generated.

iBoonie replied:
Oops, this is some leftover stuff that's not needed from some other build stuff.

iBoonie commented:
> Does this also automatically upload the builds together with releases?

This will just upload artifacts (under Actions > name of workflow > artifact at the bottom). If you want I can make it so it will automatically create a release as well (manually run is usually the best for this so its not generating a release every push).

iBoonie committed:
Cleanup

chrb22 commented:
Yeah I meant when I manually create a new tag it would make a release with the builds. I've previously had problems with compiling against a glibc version that's too new for some servers, so it would be nice if I didn't have to bother with that by having it be built for me.

chrb22 commented:
I also see that you set it to use clang, but when I tried to compile it using clang locally I got problems with the tf2 sdk. But if it works with the action then it's all good.

iBoonie committed:
Create release CI

This will also listen for v* tags and automatically build a release version

iBoonie commented:
This should (hopefully) be done now. It will listen for tags and automatically build a release version. Currently listens for v* (example v1.1), but can be configured to listen for other tags. And yep, clang works perfect, I made sure I tested that.

iBoonie committed:
Formatting

iBoonie committed:
Only use tags to build

If we want this I can add it, but if we manually build from GH, the tag names will be main and mess up a bunch of stuff.

chrb22 commented:
Good to know clang is a me problem. This will also be a nice reference for me if I want to use actions for other things.
